src/rpdk/core/contract/resource_client.py

```diff
@@ -370,7 +370,10 @@ def call(self, action, request):
             self.assert_write_only_property_does_not_exist(response["resourceModel"])
             sleep(callback_delay_seconds)
 
-            payload["callbackContext"] = response.get("callbackContext")
+            request["desiredResourceState"] = response.get("resourceModel")
+            payload = self._make_payload(
+                action, request, response.get("callbackContext")
```

Comment:
can we just make this:
```
payload["callbackContext"] = response.get("callbackContext")
request["desiredResourceState"] = response.get("resourceModel")
```
? what benefit do we get from making the payload again? i guess changing the request token?
```

Reply:
> can we just make this:
> 
> ```
> payload["callbackContext"] = response.get("callbackContext")
> request["desiredResourceState"] = response.get("resourceModel")
> ```
> 
> ? what benefit do we get from making the payload again? i guess changing the request token?
> 
> ```
> 
> ```

you are right, it makes sense to just update